### PR TITLE
Use DOCtor-RST 1.47.1

### PR DESCRIPTION
**Pull request by OskarStark**
https://github.com/OskarStark/doctor-rst/releases/tag/1.47.1

cc @alamirault @staabm

### CODE REVIEW

**Comment by staabm** [comment]
👍 

ship it :shipit:

**Comment by alamirault** [comment]
We can enable rule `NoDuplicateUseStatements`

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
                     key: ${{ runner.os }}-doctor-rst-${{ steps.extract_base_branch.outputs.branch }}
 
             -   name: "Run DOCtor-RST"
-                uses: docker://oskarstark/doctor-rst:1.45.0
+                uses: docker://oskarstark/doctor-rst:1.47.0
```
Comment (alamirault) [inline .github/workflows/ci.yaml]: We can also enable rule NoDuplicateUseStatements

**Comment by staabm** [comment]
are these reported DOCtor-RST errors regressions in the tool because of the recent changes?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (OskarStark) [reply .github/workflows/ci.yaml]: This one is failing, I created a PR to check if the regression is in 1.47 or 1.46.1 

See #18326

**Comment by OskarStark** [comment]
> are these reported DOCtor-RST errors regressions in the tool because of the recent changes? 

checking right now

**Comment by OskarStark** [comment]
It looks like, yes

**Comment by staabm** [comment]
could you narrow it to one of the PRs?

**Comment by OskarStark** [comment]
I need to check it next week

**Comment by staabm** [comment]
locally I cannot reproduce the errors. maybe we need to invalidate the CI cache?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -73,7 +73,7 @@ jobs:
                     key: ${{ runner.os }}-doctor-rst-${{ steps.extract_base_branch.outputs.branch }}
 
             -   name: "Run DOCtor-RST"
-                uses: docker://oskarstark/doctor-rst:1.46.1
+                uses: docker://oskarstark/doctor-rst:1.47.0
                 with:
                     args: --short --error-format=github --cache-file=/github/workspace/.cache/doctor-rst.cache
```
Comment (OskarStark) [inline .github/workflows/ci.yaml:78]:
```suggestion
                    args: --short --error-format=github
```

lets check

**Comment by staabm** [comment]
I was able to narrow it to this PR https://github.com/OskarStark/doctor-rst/pull/1407

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (OskarStark) [reply .github/workflows/ci.yaml]: I enabled the rule

Comment (OskarStark) [reply .github/workflows/ci.yaml]:
```suggestion
                uses: docker://oskarstark/doctor-rst:1.47.1
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -73,9 +73,9 @@ jobs:
                     key: ${{ runner.os }}-doctor-rst-${{ steps.extract_base_branch.outputs.branch }}
 
             -   name: "Run DOCtor-RST"
-                uses: docker://oskarstark/doctor-rst:1.46.1
+                uses: docker://oskarstark/doctor-rst:1.47.1
                 with:
-                    args: --short --error-format=github --cache-file=/github/workspace/.cache/doctor-rst.cache
+                    args: --short --error-format=github
```
Comment (OskarStark) [inline .github/workflows/ci.yaml]:
```suggestion
                    args: --short --error-format=github --cache-file=/github/workspace/.cache/doctor-rst.cache
```

**Review by javiereguiluz — APPROVED** [review]

**Comment by OskarStark** [comment]
Thanks @staabm for fixing the regression 👍

**Pull request merged**